backend/app/controllers/api/v1/courses_controller.rb

```diff
@@ -5,17 +5,44 @@ module V1
     class CoursesController < ApplicationController
       include Dry::Monads::Result::Mixin
 
+      def index
+        @courses = search_courses(Course.includes(:user).all.limit(100))
```

Comment:
No need to use `#all` if you're using `#limit`
Just play with it in rails console